### PR TITLE
SQLite persistence implementation

### PR DESCRIPTION
**Pull request by OmarQurashi868**
Implemented the database interface, I appended a static database context in the TextMessage classes that saves a message to the DB when `message.SaveToDb()` is called.

Also added a static function `GetMessageHistory("channel", from_id, to_id)` function that returns a list of TextMessages in the specified channel, from the `from_id` id to the `to_id` id.

[Trello ticket](https://trello.com/c/ULfg7val/43-sqlite-persistence-implementation)

### CODE REVIEW

File: Backend/Database/Database.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        {
+            optionsBuilder.UseSqlite($"Data Source=Messages.db");
+        }
+
+        public static void SetUp() 
+        {
+            TextMessageContext messageContext = new TextMessageContext();
+            // This vvv should be disabled in prod
```
Comment (ahmed-okkio) [inline Backend/Database/Database.cs:18]: should this go into main like this?

File: Backend/Models/CommandMessage.cs [hunk 2]
```diff
@@ -17,13 +15,14 @@ internal class CommandMessage : Message
 
         private CommandType ParseCommandType()
         {
+            // TODO: Implement parser and commandtypes
```
Comment (ahmed-okkio) [inline Backend/Models/CommandMessage.cs]: Remove this comment since it will conflict with Akrums PR

File: Backend/Models/Message.cs [hunk 3]
```diff
@@ -45,6 +45,16 @@ public MessageParams(string rawString)
         protected DateTime _at;
         protected string _with;
 
+        protected Message()
+        {
+            _do = String.Empty;
```
Comment (ahmed-okkio) [inline Backend/Models/Message.cs:50]: can't you intialize the values above when they are declared instead?

File: Backend/Models/TextMessage.cs [hunk 4]
```diff
@@ -1,29 +1,44 @@
...
         private string _content;
         public string Content
         {
             get { return _content; }
+            set { _content = value; }
+        }
+
+        private TextMessage ()
+        {
+            _sender = String.Empty;
```
Comment (ahmed-okkio) [inline Backend/Models/TextMessage.cs:39]: same as before

File: Backend/Database/Database.cs (hunk 1 above)
Comment (OmarQurashi868) [reply Backend/Database/Database.cs:18]: Monka my bad

File: Backend/Models/CommandMessage.cs (hunk 2 above)
Comment (OmarQurashi868) [reply Backend/Models/CommandMessage.cs]: Done

File: Backend/Database/Database.cs [hunk 5]
```diff
@@ -16,7 +16,7 @@ public static void SetUp()
         {
             TextMessageContext messageContext = new TextMessageContext();
             // This vvv should be disabled in prod
-            messageContext.Database.EnsureDeleted();
+            // messageContext.Database.EnsureDeleted();
```
Comment (ahmed-okkio) [inline Backend/Database/Database.cs:19]: What purpose does this serve? ya3ni why not remove it completely?

**Review by ahmed-okkio — APPROVED** [review]

File: Backend/Database/Database.cs (hunk 5 above)
Comment (OmarQurashi868) [reply Backend/Database/Database.cs:19]: It clears the database from messages when starting the application, useful for testing

**Pull request merged**